src/main/java/hsien/Hsien.java

```diff
@@ -0,0 +1,260 @@
+            String toDate = dates[1].substring(0, dates[1].length()-1).trim();
+            newTask = new Event(tempDesc, fromDate, toDate);
+        }
+
+        // Mark task
+        if (isMark.equals("X")) {
+            newTask.mark();
+        }
+
+        messages.add(newTask);
+        System.out.println("Added task: " + newTask.getDescription());
+        System.out.printf("Now you have [%d] tasks in the list.%n", messages.size());
+    }
+
+    public static void addTask(String command, String desc, ArrayList<Task> messages) {
+        // Empty task
+        try {
+            if (desc.isEmpty()) {
+                throw new HsienException();
+            }
+        } catch (HsienException e) {
+            System.out.println("Description cannot be left empty");
+            printLine();
+            return;
+        }
+
+        Task newTask = null;
+        String tempDesc;
+
+        // Create Task object based on action
+        if (command.equals("todo")) {
+            newTask = new Todo(desc);
+        } else if (command.equals("deadline")) {
+            tempDesc = desc.split("/by")[0].trim();
+            String byDate = desc.split("/by")[1].trim();
+            newTask = new Deadline(tempDesc, byDate);
+        } else {
+            tempDesc = desc.split("/from")[0].trim();
+            String[] dates = desc.split("/from")[1].split("/to");
+            String fromDate = dates[0].trim();
+            String toDate = dates[1].trim();
+            newTask = new Event(tempDesc, fromDate, toDate);
+        }
+
+        messages.add(newTask);
+        System.out.println("Added task: " + newTask.getDescription());
+        System.out.printf("Now you have [%d] tasks in the list.%n", messages.size());
+    }
+
+    public static void handleMark(String command, String desc, ArrayList<Task> messages) {
+        // Get the task index
+        int index = Integer.parseInt(desc);
+        boolean isMarking = command.equals("mark");
+
+        // Perform marking or unmarking
+        if (isMarking) {
+            messages.get(index - 1).mark();
+            System.out.println("You marked " + index + " as marked");
+        } else {
+            messages.get(index - 1).unmark();
+            System.out.println("You marked " + index + " as unmarked");
+        }
+
+        // Print status of the task
+        System.out.println(messages.get(index - 1).getStatusDescription());
+    }
+
+    public static void deleteTask(ArrayList<Task> messages, int index) {
+        System.out.println("Noted I have removed this task");
+        System.out.println(messages.get(index).getStatusDescription());
+        messages.remove(index);
+        System.out.printf("Now you have %d tasks in the list\n", messages.size());
+    }
+
+    public static void main(String[] args) {
+        // Initialise variables
+        List<String> validCommands = Arrays.asList("bye", "list", "mark", "unmark", "delete", "todo", "deadline", "event");
```

Comment:
Consider using a normal array instead of Arrays.asList for the validCommands since these values are predefined and will not change. This can improve performance slightly and make the intent clearer.